core/src/main/java/com/linecorp/armeria/client/HttpChannelPool.java

```diff
@@ -165,9 +165,34 @@ private static int toIndex(SerializationFormat serializationFormat) {
         return toIndex(serializationFormat == SerializationFormat.WS);
     }
 
-    private SslContext determineSslContext(SessionProtocol desiredProtocol) {
-        return desiredProtocol == SessionProtocol.H1 || desiredProtocol == SessionProtocol.H1C ?
-               sslCtxHttp1Only : sslCtxHttp1Or2;
+    private SslContext determineSslContext(SessionProtocol desiredProtocol, boolean preferHttp1) {
+        if (preferHttp1) {
+            switch (desiredProtocol) {
+                case HTTP:
+                case HTTPS:
```

Comment:
nit; Would just adding a `return preferHttp1 ? sslCtxHttp1Only : sslCtxHttp1Or2;` condition be simpler?

Reply:
`preferHttp1` does not mean HTTP/2 is disallowed. HTTP/2 could be used if `H2` or `H2C` is specified.

Reply:
But if `HTTP/HTTPS` is used, isn't HTTP/2 disallowed?

Reply:
I may not understand your suggestion. Would you illustrate the full code? 😅

Reply:
Sorry should've been more clear 😅   This was just a minor refactoring suggestion, feel free to ignore

```java
    private SslContext determineSslContext(SessionProtocol desiredProtocol, boolean preferHttp1) {
        switch (desiredProtocol) {
            case H1:
            case H1C:
                 return sslCtxHttp1Only;
            case HTTP:
            case HTTPS:
                return preferHttp1 ? sslCtxHttp1Only : sslCtxHttp1Or2;
            case H2:
            case H2C:
                return sslCtxHttp1Or2;
            default:
                throw new Error(); // Should never reach here.
        }
    }
```

Reply:
Got it. It looks better.